### ISSUE TITLE
strings to-lower does not work for non-ascii characters

### ISSUE THREAD

**Issue by jmoerschbach** [widelands/widelands#5448]
**Describe the bug**
The to_lower function https://github.com/widelands/widelands/blob/d7d22c505d947cadb62198f6a557827aeba2503b/src/base/string.cc#L23 does not work for non-ascii (utf) characters like german umlauts. This is for example visible when filtering dropdowns

**To Reproduce**
1. Language is german
2. Start creating a new random game
3. Click on clima dropdown and enter "ö"
4. No matching entry is visible 

**Expected behavior**
"Ödland" should be visible in the filtered list list

**Version:**
 - OS:  Ubuntu 22-04
 - Widelands Version: 1.1~git133 (8b1ac19@master)

**Additional context**
This particular case may be fixed by using [icu::UnicodeString](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1UnicodeString.html) as proposed in https://github.com/widelands/widelands/pull/5443#issuecomment-1149478604 but I think it could be a more general problem (need to check where to_lower is used)

**Comment by akupar** [comment]
I agree this would be better, but I think it's not worth doing if it slows the game down. I looked up all the places where to_lower is used. It doesn't seem to be used with localized texts. Here are all the uses:

`xxx/widelands$ grep -Rin to_lower ./src/`
[./src/base/string.cc:23](https://github.com/widelands/widelands/blob/master/src/base/string.cc#L23):`std::string to_lower(const std::string& str) {`
[./src/base/string.h:34](https://github.com/widelands/widelands/blob/master/src/base/string.h#L34): `std::string to_lower(const std::string&);`
[./src/base/test/test_string.cc:36](https://github.com/widelands/widelands/blob/master/src/base/test/test_string.cc#L36): `check_equal(to_lower(str2), str2);`
[./src/base/test/test_string.cc:37](https://github.com/widelands/widelands/blob/master/src/base/test/test_string.cc#L37): `check_equal(to_lower(str1), str2);`
[./src/base/test/test_string.cc:38](https://github.com/widelands/widelands/blob/master/src/base/test/test_string.cc#L38): `check_equal(to_lower(str3) == str2, false);`

--

[./src/base/string.h:49](https://github.com/widelands/widelands/blob/master/src/base/string.h#L49): `return (case_sensitive ? str.find(test) : to_lower(str).find(to_lower(test))) !=`

Used in the function `contains` when it is called with `case_insensitive=true` parameter, but contains is never called with it except in some tests.

--

[./src/logic/map.cc:2124](https://github.com/widelands/widelands/blob/master/src/logic/map.cc#L2124): `const std::string lower_filename = to_lower(filename);`

This code is only interested in the extension which is probably always ascii.

--

[./src/logic/map_objects/immovable.cc:201](https://github.com/widelands/widelands/blob/master/src/logic/map_objects/immovable.cc#L201): `program_name = to_lower(program_name);`
[./src/logic/map_objects/immovable.cc:580](https://github.com/widelands/widelands/blob/master/src/logic/map_objects/immovable.cc#L580): `program_name = to_lower(program_name);`
[./src/logic/map_objects/tribes/productionsite.cc:166](https://github.com/widelands/widelands/blob/master/src/logic/map_objects/tribes/productionsite.cc#L166): `program_name = to_lower(program_name);`
[./src/logic/map_objects/tribes/worker_descr.cc:104](https://github.com/widelands/widelands/blob/master/src/logic/map_objects/tribes/worker_descr.cc#L104): `program_name = to_lower(program_name);`
[./src/map_io/map_buildingdata_packet.cc:763](https://github.com/widelands/widelands/blob/master/src/map_io/map_buildingdata_packet.cc#L763): `program_name = to_lower(program_name);`

Not sure about these. I think these are some internal strings that are not localized.

--

[./src/network/internet_gaming.cc:1053](https://github.com/widelands/widelands/blob/master/src/network/internet_gaming.cc#L1053): `return to_lower(username) != "team";`

Compared to static string.

--

[./src/ui_basic/dropdown.h:320](https://github.com/widelands/widelands/blob/master/src/ui_basic/dropdown.h#L320): `const std::string lowered_input_text = to_lower(input_text);`
[./src/ui_basic/dropdown.h:407](https://github.com/widelands/widelands/blob/master/src/ui_basic/dropdown.h#L407): `if (input_text == to_lower(x.hotkey) && is_in_filtered_list(x.value)) {`
